samcli/lib/pipeline/bootstrap/resource.py

```diff
@@ -0,0 +1,133 @@
+""" Represents AWS resource"""
+from typing import List, Optional
+
+
+class ARNParts:
```

Comment:
this class can be converted into a `NamedTuple` because we don't need to mutate any attributes after it is created.

Reply:
seems like an invalid NamedTupel case; "Cannot overwrite NamedTuple attribute __init__"